amg/inc/adapt.hpp

```diff
@@ -127,6 +128,31 @@ int adapt_approx_xbad(
     double *err, double *err0, int *executed_iters, double rtol/*=10e-12*/, double atol/*=10e-24*/,
     bool normalize/*=1*/, bool output=true);
 
+/*! \brief Updates operators and relaxation.
+
+    It updates the weighted l1-smoother in accordance to the new matrix \a A.
```

Comment:
I feel like the documentation of this function and its multilevel sister could be more clear. These descriptions suggest the main thing that is changing is the *smoother*, where really the whole hierarchy changes. I recommend something like "Recomputes hierarchy while assuming spectral information is the same."

Reply:
Done